test/chpldoc/globals/variablePrecedence.doc.chpl

```diff
@@ -0,0 +1,104 @@
+module variablePrecedence {
+  var a, b, c, d, e, f: int;
+
+  // Simple cases handled in other tests
+
+  var pp = a+b+c;
+  var Pp = (a+b)+c;
+  var pP = a+(b+c);
+
+  var pppp = a+b+c+d+e;
+  var PpPp = (a+b)+(c+(d+e));
+
+  var mm = a-b-c;
+  var Mm = (a-b)-c;
+  var mM = a-(b-c);
+
+  var pm = a+b-c;
+  var Pm = (a+b)-c;
+  var pM = a+(b-c);
+
+  var mp = a-b+c;
+  var Mp = (a-b)+c;
+  var mP = a-(b+c);
+
+  // Writing with parens to avoid the compiler warning
+  var unarypunaryp = +(+a);
+  var unarymunarym = -(-a);
+
+  var unarypunarym = +-a;
+  var unarymunaryp = -+a;
+
+  var unarypP = +(a+b);
+  var unarypM = +(a-b);
+  var unarymP = -(a+b);
+  var unarymM = -(a-b);
+
+  var unarynp = ~(a+b);
+
+  var tp = a*b+c;
+  var Tp = (a*b)+c;
+  var tP = a*(b+c);
+
+  var pt = a+b*c;
+  var Pt = (a+b)*c;
+  var pT = a+(b*c);
+
+  var td = a*b/c;
+  var Td = (a*b)/c;
+  var tD = a*(b/c);
+
+  var dt = a/b*c;
+  var Dt = (a/b)*c;
+  var dT = a/(b*c);
+
+  var ll = a<<b<<c;
+  var Ll = (a<<b)<<c;
+  var lL = a<<(b<<c);
+
+  var ee = a == b == c;
+  var Ee = (a == b) == c;
+  var eE = a == (b == c);
+
+  // ** is right-associative
+  var xx = a**b**c;
+  var xX = a**(b**c);
+  var Xx = (a**b)**c;
+
+  var xpx = a**b + c**d;
+  var xPx = a**(b+c)**d;
+  var xPX = a**(b+c**d);
+
+  var ooo = a || b || c || d;
+  var oao = a || b && c || d;
```

Comment:
The spacing on this case makes it a little hard to read in the generated chpldoc output, should we improve it?

Reply:
My thinking is that all of the binary operator cases would look better with spaces between the operator itself and the arguments to it.  This is just the case that really made it hard for me to read.

I don't think there needs to be much done to determine when spaces should be added or not, there just should be a space in between the binary operator itself and each of the arguments.

Reply:
I can give that a try and see how it looks globally.

Reply:
I added the spaces in the update I just pushed.

I did **not** update the .good files, since there were a half dozen with changes, and I'm on the fence as to whether it's an overall improvement.  If you prefer it with all the spaces, I'll certainly do that.

I'm ok with this not making the release, especially since apart from the spaces here, the whole PR doesn't result in any `make docs` changes.  That's why I'm waiting for your review of the changed output before taking the time to do that.

The aforementioned `make docs` changes from these spaces:

<details>
<summary><tt>make docs</tt></summary>
<pre>
diff -r doc-master/html/builtins/Bytes.html doc/html/builtins/Bytes.html
396c396
< <em class="property">proc </em><code class="sig-name descname">createBytesWithNewBuffer</code><span class="sig-paren">(</span><em class="sig-param">x: c_ptr(?t)</em>, <em class="sig-param">length: int</em>, <em class="sig-param">size = length+1</em><span class="sig-paren">)</span></dt>
---
> <em class="property">proc </em><code class="sig-name descname">createBytesWithNewBuffer</code><span class="sig-paren">(</span><em class="sig-param">x: c_ptr(?t)</em>, <em class="sig-param">length: int</em>, <em class="sig-param">size = length + 1</em><span class="sig-paren">)</span></dt>
diff -r doc-master/html/builtins/String.html doc/html/builtins/String.html
501c501
< <em class="property">proc </em><code class="sig-name descname">createStringWithNewBuffer</code><span class="sig-paren">(</span><em class="sig-param">x: c_ptr(?t)</em>, <em class="sig-param">length: int</em>, <em class="sig-param">size = length+1</em>, <em class="sig-param">policy = decodePolicy.strict</em><span class="sig-paren">)</span> throws</dt>
---
> <em class="property">proc </em><code class="sig-name descname">createStringWithNewBuffer</code><span class="sig-paren">(</span><em class="sig-param">x: c_ptr(?t)</em>, <em class="sig-param">length: int</em>, <em class="sig-param">size = length + 1</em>, <em class="sig-param">policy = decodePolicy.strict</em><span class="sig-paren">)</span> throws</dt>
diff -r doc-master/html/modules/packages/DistributedBag.html doc/html/modules/packages/DistributedBag.html
332c332
< <em class="property">config const </em><code class="sig-name descname">distributedBagMaxBlockSize</code> = 1024*1024<a class="headerlink" href="#DistributedBag.distributedBagMaxBlockSize" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">config const </em><code class="sig-name descname">distributedBagMaxBlockSize</code> = 1024 * 1024<a class="headerlink" href="#DistributedBag.distributedBagMaxBlockSize" title="Permalink to this definition">¶</a></dt>
diff -r doc-master/html/modules/packages/HDF5/C_HDF5.html doc/html/modules/packages/HDF5/C_HDF5.html
8232c8232
< <em class="property">param </em><code class="sig-name descname">H5F_OBJ_ALL</code> = H5F_OBJ_FILE|H5F_OBJ_DATASET|H5F_OBJ_GROUP|H5F_OBJ_DATATYPE|H5F_OBJ_ATTR<a class="headerlink" href="#C_HDF5.H5F_OBJ_ALL" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">param </em><code class="sig-name descname">H5F_OBJ_ALL</code> = H5F_OBJ_FILE | H5F_OBJ_DATASET | H5F_OBJ_GROUP | H5F_OBJ_DATATYPE | H5F_OBJ_ATTR<a class="headerlink" href="#C_HDF5.H5F_OBJ_ALL" title="Permalink to this definition">¶</a></dt>
diff -r doc-master/html/modules/standard/GMP.html doc/html/modules/standard/GMP.html
340c340
< <em class="property">type </em><code class="sig-name descname">mpz_t</code> = 1*__mpz_struct<a class="headerlink" href="#GMP.mpz_t" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">type </em><code class="sig-name descname">mpz_t</code> = 1 * __mpz_struct<a class="headerlink" href="#GMP.mpz_t" title="Permalink to this definition">¶</a></dt>
346c346
< <em class="property">type </em><code class="sig-name descname">mpf_t</code> = 1*__mpf_struct<a class="headerlink" href="#GMP.mpf_t" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">type </em><code class="sig-name descname">mpf_t</code> = 1 * __mpf_struct<a class="headerlink" href="#GMP.mpf_t" title="Permalink to this definition">¶</a></dt>
352c352
< <em class="property">type </em><code class="sig-name descname">gmp_randstate_t</code> = 1*__gmp_randstate_struct<a class="headerlink" href="#GMP.gmp_randstate_t" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">type </em><code class="sig-name descname">gmp_randstate_t</code> = 1 * __gmp_randstate_struct<a class="headerlink" href="#GMP.gmp_randstate_t" title="Permalink to this definition">¶</a></dt>
diff -r doc-master/html/modules/standard/IO.html doc/html/modules/standard/IO.html
2060c2060
< <em class="property">proc </em><code class="sig-prename descclassname">channel.</code><code class="sig-name descname">readline</code><span class="sig-paren">(</span><em class="sig-param">arg: [] uint(8)</em>, <em class="sig-param">out numRead: int</em>, <em class="sig-param">start = arg.domain.low</em>, <em class="sig-param">amount = arg.domain.high-start+1</em><span class="sig-paren">)</span>: bool throws<a class="headerlink" href="#IO.channel.readline" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">proc </em><code class="sig-prename descclassname">channel.</code><code class="sig-name descname">readline</code><span class="sig-paren">(</span><em class="sig-param">arg: [] uint(8)</em>, <em class="sig-param">out numRead: int</em>, <em class="sig-param">start = arg.domain.low</em>, <em class="sig-param">amount = arg.domain.high - start + 1</em><span class="sig-paren">)</span>: bool throws<a class="headerlink" href="#IO.channel.readline" title="Permalink to this definition">¶</a></dt>
diff -r doc-master/html/modules/standard/Math.html doc/html/modules/standard/Math.html
371c371
< <em class="property">proc </em><code class="sig-name descname">abs</code><span class="sig-paren">(</span><em class="sig-param">z: complex(?w)</em><span class="sig-paren">)</span>: real(w/2)</dt>
---
> <em class="property">proc </em><code class="sig-name descname">abs</code><span class="sig-paren">(</span><em class="sig-param">z: complex(?w)</em><span class="sig-paren">)</span>: real(w / 2)</dt>
382c382
< <em class="property">proc </em><code class="sig-name descname">carg</code><span class="sig-paren">(</span><em class="sig-param">z: complex(?w)</em><span class="sig-paren">)</span>: real(w/2)<a class="headerlink" href="#Math.carg" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">proc </em><code class="sig-name descname">carg</code><span class="sig-paren">(</span><em class="sig-param">z: complex(?w)</em><span class="sig-paren">)</span>: real(w / 2)<a class="headerlink" href="#Math.carg" title="Permalink to this definition">¶</a></dt>
625c625
< <em class="property">proc </em><code class="sig-name descname">cproj</code><span class="sig-paren">(</span><em class="sig-param">z: complex(?w)</em><span class="sig-paren">)</span>: real(w/2)<a class="headerlink" href="#Math.cproj" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">proc </em><code class="sig-name descname">cproj</code><span class="sig-paren">(</span><em class="sig-param">z: complex(?w)</em><span class="sig-paren">)</span>: real(w / 2)<a class="headerlink" href="#Math.cproj" title="Permalink to this definition">¶</a></dt>
diff -r doc-master/html/modules/standard/Random.html doc/html/modules/standard/Random.html
286c286
< <em class="property">type </em><code class="sig-name descname">RandomStream</code> = if defaultRNG==RNG.PCG then PCGRandomStream else NPBRandomStream<a class="headerlink" href="#Random.RandomStream" title="Permalink to this definition">¶</a></dt>
---
> <em class="property">type </em><code class="sig-name descname">RandomStream</code> = if defaultRNG == RNG.PCG then PCGRandomStream else NPBRandomStream<a class="headerlink" href="#Random.RandomStream" title="Permalink to this definition">¶</a></dt>
diff -r doc-master/html/_sources/builtins/Bytes.rst.txt doc/html/_sources/builtins/Bytes.rst.txt
185c185
< .. function:: proc createBytesWithNewBuffer(x: c_ptr(?t), length: int, size = length+1)
---
> .. function:: proc createBytesWithNewBuffer(x: c_ptr(?t), length: int, size = length + 1)
diff -r doc-master/html/_sources/builtins/String.rst.txt doc/html/_sources/builtins/String.rst.txt
305c305
< .. function:: proc createStringWithNewBuffer(x: c_ptr(?t), length: int, size = length+1, policy = decodePolicy.strict) throws
---
> .. function:: proc createStringWithNewBuffer(x: c_ptr(?t), length: int, size = length + 1, policy = decodePolicy.strict) throws
diff -r doc-master/html/_sources/modules/packages/DistributedBag.rst.txt doc/html/_sources/modules/packages/DistributedBag.rst.txt
116c116
< .. data:: config const distributedBagMaxBlockSize = 1024*1024
---
> .. data:: config const distributedBagMaxBlockSize = 1024 * 1024
diff -r doc-master/html/_sources/modules/packages/HDF5/C_HDF5.rst.txt doc/html/_sources/modules/packages/HDF5/C_HDF5.rst.txt
3246c3246
< .. data:: param H5F_OBJ_ALL = H5F_OBJ_FILE|H5F_OBJ_DATASET|H5F_OBJ_GROUP|H5F_OBJ_DATATYPE|H5F_OBJ_ATTR
---
> .. data:: param H5F_OBJ_ALL = H5F_OBJ_FILE | H5F_OBJ_DATASET | H5F_OBJ_GROUP | H5F_OBJ_DATATYPE | H5F_OBJ_ATTR
diff -r doc-master/html/_sources/modules/standard/GMP.rst.txt doc/html/_sources/modules/standard/GMP.rst.txt
132c132
< .. type:: type mpz_t = 1*__mpz_struct
---
> .. type:: type mpz_t = 1 * __mpz_struct
136c136
< .. type:: type mpf_t = 1*__mpf_struct
---
> .. type:: type mpf_t = 1 * __mpf_struct
140c140
< .. type:: type gmp_randstate_t = 1*__gmp_randstate_struct
---
> .. type:: type gmp_randstate_t = 1 * __gmp_randstate_struct
diff -r doc-master/html/_sources/modules/standard/IO.rst.txt doc/html/_sources/modules/standard/IO.rst.txt
1686c1686
< .. method:: proc channel.readline(arg: [] uint(8), out numRead: int, start = arg.domain.low, amount = arg.domain.high-start+1): bool throws
---
> .. method:: proc channel.readline(arg: [] uint(8), out numRead: int, start = arg.domain.low, amount = arg.domain.high - start + 1): bool throws
diff -r doc-master/html/_sources/modules/standard/Math.rst.txt doc/html/_sources/modules/standard/Math.rst.txt
125c125
< .. function:: proc abs(z: complex(?w)): real(w/2)
---
> .. function:: proc abs(z: complex(?w)): real(w / 2)
132c132
< .. function:: proc carg(z: complex(?w)): real(w/2)
---
> .. function:: proc carg(z: complex(?w)): real(w / 2)
311c311
< .. function:: proc cproj(z: complex(?w)): real(w/2)
---
> .. function:: proc cproj(z: complex(?w)): real(w / 2)
diff -r doc-master/html/_sources/modules/standard/Random.rst.txt doc/html/_sources/modules/standard/Random.rst.txt
82c82
< .. type:: type RandomStream = if defaultRNG==RNG.PCG then PCGRandomStream else NPBRandomStream
---
> .. type:: type RandomStream = if defaultRNG == RNG.PCG then PCGRandomStream else NPBRandomStream
diff -r doc-master/rst/builtins/Bytes.rst doc/rst/builtins/Bytes.rst
185c185
< .. function:: proc createBytesWithNewBuffer(x: c_ptr(?t), length: int, size = length+1)
---
> .. function:: proc createBytesWithNewBuffer(x: c_ptr(?t), length: int, size = length + 1)
diff -r doc-master/rst/builtins/String.rst doc/rst/builtins/String.rst
305c305
< .. function:: proc createStringWithNewBuffer(x: c_ptr(?t), length: int, size = length+1, policy = decodePolicy.strict) throws
---
> .. function:: proc createStringWithNewBuffer(x: c_ptr(?t), length: int, size = length + 1, policy = decodePolicy.strict) throws
diff -r doc-master/rst/modules/packages/DistributedBag.rst doc/rst/modules/packages/DistributedBag.rst
116c116
< .. data:: config const distributedBagMaxBlockSize = 1024*1024
---
> .. data:: config const distributedBagMaxBlockSize = 1024 * 1024
diff -r doc-master/rst/modules/packages/HDF5/C_HDF5.rst doc/rst/modules/packages/HDF5/C_HDF5.rst
3246c3246
< .. data:: param H5F_OBJ_ALL = H5F_OBJ_FILE|H5F_OBJ_DATASET|H5F_OBJ_GROUP|H5F_OBJ_DATATYPE|H5F_OBJ_ATTR
---
> .. data:: param H5F_OBJ_ALL = H5F_OBJ_FILE | H5F_OBJ_DATASET | H5F_OBJ_GROUP | H5F_OBJ_DATATYPE | H5F_OBJ_ATTR
diff -r doc-master/rst/modules/standard/GMP.rst doc/rst/modules/standard/GMP.rst
132c132
< .. type:: type mpz_t = 1*__mpz_struct
---
> .. type:: type mpz_t = 1 * __mpz_struct
136c136
< .. type:: type mpf_t = 1*__mpf_struct
---
> .. type:: type mpf_t = 1 * __mpf_struct
140c140
< .. type:: type gmp_randstate_t = 1*__gmp_randstate_struct
---
> .. type:: type gmp_randstate_t = 1 * __gmp_randstate_struct
diff -r doc-master/rst/modules/standard/IO.rst doc/rst/modules/standard/IO.rst
1686c1686
< .. method:: proc channel.readline(arg: [] uint(8), out numRead: int, start = arg.domain.low, amount = arg.domain.high-start+1): bool throws
---
> .. method:: proc channel.readline(arg: [] uint(8), out numRead: int, start = arg.domain.low, amount = arg.domain.high - start + 1): bool throws
diff -r doc-master/rst/modules/standard/Math.rst doc/rst/modules/standard/Math.rst
125c125
< .. function:: proc abs(z: complex(?w)): real(w/2)
---
> .. function:: proc abs(z: complex(?w)): real(w / 2)
132c132
< .. function:: proc carg(z: complex(?w)): real(w/2)
---
> .. function:: proc carg(z: complex(?w)): real(w / 2)
311c311
< .. function:: proc cproj(z: complex(?w)): real(w/2)
---
> .. function:: proc cproj(z: complex(?w)): real(w / 2)
diff -r doc-master/rst/modules/standard/Random.rst doc/rst/modules/standard/Random.rst
82c82
< .. type:: type RandomStream = if defaultRNG==RNG.PCG then PCGRandomStream else NPBRandomStream
---
> .. type:: type RandomStream = if defaultRNG == RNG.PCG then PCGRandomStream else NPBRandomStream
Binary files doc-master/util/__pycache__/chpl2rst.cpython-38.pyc and doc/util/__pycache__/chpl2rst.cpython-38.pyc differ
Binary files doc-master/util/__pycache__/disguise.cpython-38.pyc and doc/util/__pycache__/disguise.cpython-38.pyc differ
Binary files doc-master/util/__pycache__/__init__.cpython-38.pyc and doc/util/__pycache__/__init__.cpython-38.pyc differ
</pre>
</details>

Reply:
I think that's much improved but since you're uncertain about whether it is better, I probably should ask others on the team to weigh in before merging.

Reply:
Do you have specific cases where you don't like the added spacing?

Reply:
In the light of day, most of them do look better with spaces.  :)

I'm still a little iffy on the exponentiation ones in this file.

In the `make docs` output, spaces in function arg default values looks a little off,

```chapel
proc channel.readline(arg: [] uint(8), out numRead: int, start = arg.domain.low, amount = arg.domain.high - start + 1)
```
And the changes of tuple-type notation from `1*__mpz_struct` to `1 * __mpz_struct` "looks less" like a tuple, and some of the changes of `real(w/2)` to `real(w / 2)` seem a little questionable.  But that may all just be due to being used to the current look.

Would it help you gather input if I did push updates to at least some of the .good files?  I don't think I can get to all of them before work.

Reply:
I think it would help, yeah!  But let's wait to do that until you've split it into a separate branch.  Then I'll point people to the PR itself and they can take a look

Reply:
Ok, I reverted the spaces and added the missing DCO.

Should I open a PR for the spacing changes, or start with an issue?

Reply:
Good question!  Let's start with an issue, listing the changes to `make docs` and the ones you're uncertain about